### PR TITLE
refactor(fold): style and formatting

### PR DESCRIPTION
**Pull request by lewis6991**
- refactor(fold): update comment formatting
- refactor(fold): use post-increment/decrement
- refactor(fold): reduce scope of local variables
- refactor(fold): use booleans

### CODE REVIEW

File: src/nvim/globals.h [hunk 1]
```diff
@@ -725,7 +725,7 @@ EXTERN bool need_highlight_changed INIT(= true);
 EXTERN FILE *scriptout INIT(= NULL);  ///< Stream to write script to.
 
 // volatile because it is used in a signal handler.
-EXTERN volatile int got_int INIT(= false);  // set to true when interrupt
+EXTERN volatile bool got_int INIT(= false); // set to true when interrupt
                                             // signal occurred
```
Comment (justinmk) [inline src/nvim/globals.h:729]:
```suggestion
EXTERN volatile bool got_int INIT(= false); // set to true when interrupt signal occurred
```

Comment (zeertzjq) [reply src/nvim/globals.h:729]: I guess this `volatile` is also no longer needed as Nvim just terminates on SIGINT.

File: src/nvim/globals.h [hunk 2]
```diff
@@ -725,7 +725,7 @@ EXTERN bool need_highlight_changed INIT(= true);
 EXTERN FILE *scriptout INIT(= NULL);  ///< Stream to write script to.
 
 // volatile because it is used in a signal handler.
-EXTERN volatile int got_int INIT(= false);  // set to true when interrupt
+EXTERN volatile bool got_int INIT(= false); // set to true when interrupt
```
[the remark this answers is not in the document]
Comment (zeertzjq) [reply src/nvim/globals.h:728]: It doesn't seem used in a signal handler any more?

[the remark this answers is not in the document]
Comment (gpanders) [reply src/nvim/globals.h:728]: If that's the case then disregard my comment :)

File: src/nvim/globals.h [hunk 3]
```diff
@@ -224,7 +224,7 @@ EXTERN dict_T vimvardict;                   // Dictionary with v: variables
 EXTERN dict_T globvardict;                  // Dictionary with g: variables
 /// g: value
 #define globvarht globvardict.dv_hashtab
-EXTERN int did_emsg;                        // set by emsg() when the message
+EXTERN bool did_emsg;                       // set by emsg() when the message
```
Comment (zeertzjq) [inline src/nvim/globals.h:227]: Why did you change this to `bool`???

Comment (lewis6991) [reply src/nvim/globals.h:227]: Is it not a bool?

Comment (zeertzjq) [reply src/nvim/globals.h:227]: No.

Comment (lewis6991) [reply src/nvim/globals.h:227]: Why not?

Comment (zeertzjq) [reply src/nvim/globals.h:227]: Because it can be larger than 1.

Comment (zeertzjq) [reply src/nvim/globals.h:227]: Search for `did_emsg_before` in `eval.c`.

Comment (lewis6991) [reply src/nvim/globals.h:227]: `did_emsg_before` is defined as an `int` but only used as a `bool`

Comment (lewis6991) [reply src/nvim/globals.h:227]: It is assigned from `did_emsg` which (from as far as I can tell) is only ever `true` or `false`.

Comment (zeertzjq) [reply src/nvim/globals.h:227]: In message.c code you deleted in this PR.

Comment (clason) [reply src/nvim/globals.h:227]: There's a few stray `int`s in `eval.c` that are derived from `did_emsg` (`save_did_emsg`, `err`, `did_emsg_before`) that probably should be `bool`, too?

Comment (lewis6991) [reply src/nvim/globals.h:227]: Ok, I see. I have no idea why I did that. We can revert that specific change?

Is this causing any issues with anything?

Comment (zeertzjq) [reply src/nvim/globals.h:227]: As I said, they must be `int`, not `bool`.

Comment (lewis6991) [reply src/nvim/globals.h:227]: #20580

Comment (clason) [reply src/nvim/globals.h:227]: Why? What precisely broke now that they can only take values `0` and `1`?

Comment (zeertzjq) [reply src/nvim/globals.h:227]: It can cause duplicate error messages in expression parsing, and it is causing a unit test failure in #20574.
